modules/renter/hostdb/hostweight.go

```diff
@@ -47,6 +63,29 @@ func calculateHostWeight(entry hostEntry) (weight types.Currency) {
 		weight = baseWeight.Div(totalPrice).Div(totalPrice).Div(totalPrice).Div(totalPrice).Div(totalPrice)
 	}
 
+	// Enact penalties if the host does not have very much storage remaining.
+	if entry.RemainingStorage < 200*requiredStorage {
+		weight = weight.Div64(2) // 2x total penalty
+	}
+	if entry.RemainingStorage < 100*requiredStorage {
+		weight = weight.Div64(3) // 6x total penalty
+	}
+	if entry.RemainingStorage < 50*requiredStorage {
+		weight = weight.Div64(4) // 24x total penalty
+	}
+	if entry.RemainingStorage < 25*requiredStorage {
+		weight = weight.Div64(5) // 95x total penalty
+	}
+	if entry.RemainingStorage < 10*requiredStorage {
+		weight = weight.Div64(6) // 570x total penalty
+	}
+	if entry.RemainingStorage < 5*requiredStorage {
+		weight = weight.Div64(10) // 5700x total penalty
+	}
+	if entry.RemainingStorage < requiredStorage {
+		weight = types.NewCurrency64(1) // Worthelss host.
```

Comment:
typo

there's probably a cleaner function for this weighting, but maybe not worth worrying about right now